server/models/user.rb

```diff
@@ -33,4 +33,9 @@ def password_matches?(raw_password)
   def self.public_attrs
     [:username, :id]
   end
+  def avatar_url
+    hash = Digest::MD5.hexdigest(if self.email then self.email.strip.downcase else "" end)
+    self.avatar ? "/api/auth/users/#{self.id}/avatar"
```

Comment:
Purely nit-picking, but I think this could be written a little more clearly with an if/else (since `hash` is only used in the fall-through):

```ruby
if self.avatar
  "/api/auth/users/#{self.id}/avatar"
else
  hash = Digest::MD5.hexdigest(self.email ? self.email.strip.downcase : "")
  "https://secure.gravatar.com/avatar/#{hash}?s=155&d=identicon"
end
```